src/main/java/baritone/pathing/movement/movements/MovementTraverse.java

```diff
@@ -152,86 +153,86 @@ public MovementState updateState(MovementState state) {
                 return state;
             }
             // and if it's fine to walk into the blocks in front
-            if (MovementHelper.avoidWalkingInto(BlockStateInterface.get(positionsToBreak[0]).getBlock())) {
+            if (MovementHelper.avoidWalkingInto(BlockStateInterface.get(ctx, positionsToBreak[0]).getBlock())) {
                 return state;
             }
-            if (MovementHelper.avoidWalkingInto(BlockStateInterface.get(positionsToBreak[1]).getBlock())) {
+            if (MovementHelper.avoidWalkingInto(BlockStateInterface.get(ctx, positionsToBreak[1]).getBlock())) {
                 return state;
             }
             // and we aren't already pressed up against the block
-            double dist = Math.max(Math.abs(player().posX - (dest.getX() + 0.5D)), Math.abs(player().posZ - (dest.getZ() + 0.5D)));
+            double dist = Math.max(Math.abs(ctx.player().posX - (dest.getX() + 0.5D)), Math.abs(ctx.player().posZ - (dest.getZ() + 0.5D)));
             if (dist < 0.83) {
                 return state;
             }
 
             // combine the yaw to the center of the destination, and the pitch to the specific block we're trying to break
             // it's safe to do this since the two blocks we break (in a traverse) are right on top of each other and so will have the same yaw
-            float yawToDest = RotationUtils.calcRotationFromVec3d(playerHead(), VecUtils.calculateBlockCenter(dest)).getYaw();
+            float yawToDest = RotationUtils.calcRotationFromVec3d(ctx.playerHead(), VecUtils.calculateBlockCenter(dest)).getYaw();
             float pitchToBreak = state.getTarget().getRotation().get().getPitch();
 
             state.setTarget(new MovementState.MovementTarget(new Rotation(yawToDest, pitchToBreak), true));
-            return state.setInput(InputOverrideHandler.Input.MOVE_FORWARD, true);
+            return state.setInput(Input.MOVE_FORWARD, true);
         }
 
         //sneak may have been set to true in the PREPPING state while mining an adjacent block
-        state.setInput(InputOverrideHandler.Input.SNEAK, false);
+        state.setInput(Input.SNEAK, false);
 
-        Block fd = BlockStateInterface.get(src.down()).getBlock();
+        Block fd = BlockStateInterface.get(ctx, src.down()).getBlock();
         boolean ladder = fd instanceof BlockLadder || fd instanceof BlockVine;
-        IBlockState pb0 = BlockStateInterface.get(positionsToBreak[0]);
-        IBlockState pb1 = BlockStateInterface.get(positionsToBreak[1]);
+        IBlockState pb0 = BlockStateInterface.get(ctx, positionsToBreak[0]);
+        IBlockState pb1 = BlockStateInterface.get(ctx, positionsToBreak[1]);
```

Comment:
I do not understand, why do we sometimes do ctx.get, ctx.bsi().get0...
wait 
i just remembered that ctx can be either IPlayerContext or CalculationContext

vomit vomit vomit!

Reply:
Yeahhh. All usages of ``ctx`` should be ``IPlayerContext``, so that shoullddddd be fine.